addon/components/ember-table-loading-more/component.js

```diff
@@ -0,0 +1,175 @@
+import { or, readOnly } from '@ember/object/computed';
+import { next, run, schedule, scheduleOnce } from '@ember/runloop';
+
+/**
+  Renders a custom loading indicator beneath the table body. Can be used to
+  implement an infinite scroll pattern. Provides optional centering to keep the
+  user-provided block centered horizontally in the scroll viewport.
+
+  ```hbs
+  <EmberTable as |t|>
+    <t.head @columns={{this.columns}} />
+    <t.body @rows={{this.rows}} />
+
+    <t.loadingMore
+      @isLoading={{this.isLoadingMore}}
+      @canLoadMore={{this.canLoadMore}}
+      @center={{true}}>
+
+      {{!-- custom spinner --}}
+      <img class="spinner" src="spinner.gif"/>
+    </t.loadingMore>
+  </EmberTable>
+  ```
+
+  @class {{ember-table-loading-more}}
+  @public
+*/
+export default Component.extend({
+  classNames: ['ember-table-loading-more'],
+
+  'data-test-ember-table-loading-more': true,
+
+  unwrappedApi: or('api.api', 'api'),
+  scrollElement: readOnly('unwrappedApi.columnTree.container'),
+
+  /**
+   * Boolean flag specifying if additional rows are being loaded. If true,
+   * indicator block will be visible below the table body.
+   *
+   * @argument isLoading
+   * @type boolean
+   */
+  isLoading: false,
+
+  /**
+   * Boolean flag specifying if there are more rows yet to load. If false, the
+   * indicator block will be removed from the DOM.
+   *
+   * @argument canLoadMore
+   * @type boolean
+   */
+  canLoadMore: true,
+
+  /**
+   * Boolean flag specifying if the indicator block should be centered
+   * horizontally in the scroll viewport.
+   *
+   * @argument center
+   * @type boolean
+   */
+  center: true,
+
+  init() {
+    this._super(...arguments);
+
+    // wrapped in `run` for testing compatibility in ember 2.12, 2.18
+    this._updateTransform = () => run(() => scheduleOnce('afterRender', this, 'updateTransform'));
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    // ignore the first run of `didReceiveAttrs` before element exists
+    if (!this.element) {
+      return;
+    }
+
+    // no observers here, no sir.
+    let canLoadMore = this.get('canLoadMore');
+    if (canLoadMore !== this._canLoadMore) {
+      scheduleOnce('afterRender', this, 'canLoadMoreChanged');
+      this._canLoadMore = canLoadMore;
+    }
+
+    let isLoading = this.get('isLoading');
+    if (isLoading !== this._isLoading) {
+      scheduleOnce('afterRender', this, 'isLoadingChanged');
+      this._isLoading = isLoading;
+    }
+
+    let center = this.get('center');
+    if (center !== this._center) {
+      scheduleOnce('afterRender', this, 'centerChanged');
+      this._center = center;
+    }
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.canLoadMoreChanged();
+    this.isLoadingChanged();
+    this.centerChanged();
+  },
+
+  willDestroyElement() {
+    if (this.get('center')) {
+      this.removeListeners();
+    }
+    this._super(...arguments);
```

Comment:
Is there a reason to call `super` last?

Reply:
It's an old-school LIFO/OO type thing. When initializing, we want to call `super` first to allow the parent class to do its setup before the child. When tearing down, we do the reverse. In this case it doesn't make any difference, but generally speaking, the parent might tear down some things that are needed for the child to do its (more specific) cleanup.